spark/core.py

```diff
@@ -310,32 +314,118 @@ def clear_rect(self, *args):
 
     # Draws circle at given coordinates
     def circle(self, *args):
-        self.check_coords("circle", *args, width_only=True)
-        arc_args = self.arc_args(*args)
-        self.canvas.fill_arc(*arc_args)
-        self.canvas.stroke_arc(*arc_args)
+        self.check_coords("circle", *args, width_only = True)
+        self.ellipse(*args, args[2])
 
     # Draws filled circle
     def fill_circle(self, *args):
-        self.check_coords("fill_circle", *args, width_only=True)
-        arc_args = self.arc_args(*args)
-        self.canvas.fill_arc(*arc_args)
+        self.check_coords("fill_circle", *args, width_only = True)
+        self.fill_ellipse(*args, args[2])
 
     # Draws circle stroke
     def stroke_circle(self, *args):
-        self.check_coords("stroke_circle", *args, width_only=True)
-        arc_args = self.arc_args(*args)
-        self.canvas.stroke_arc(*arc_args)
-        
+        self.check_coords("stroke_circle", *args, width_only = True)
+        self.fill_ellipse(*args, args[2])
```

Comment:
Is this supposed to be `stroke_ellipse`?
```suggestion
        self.stroke_ellipse(*args, args[2])
```

Reply:
Yep, good catch! A failure on my part to not check the individual circle functions thoroughly.